relay-server/src/actors/processor.rs

```diff
@@ -2689,7 +2689,16 @@ impl EnvelopeProcessorService {
             self.normalize_dsc(state);
             self.filter_event(state)?;
             self.run_dynamic_sampling(state);
-            self.extract_metrics(state)?;
+
+            let is_processing =
+                cfg!(feature = "processing") && self.inner.config.processing_enabled();
+
+            // We avoid extracting metrics if not sampling on pops, in order to synchronize
+            // rate limits on indexed and processed transactions.
+            if is_processing || state.sampling_result.should_drop() {
```

Comment:
If we were strict, you should only skip this on version `3`. However, this also performs correctly if you unconditionally skip here, so I'm +1 on the simplicity.